src/shell/filesystem_shell.rs

```diff
@@ -144,10 +143,10 @@ impl Shell for FilesystemShell {
                         if let Ok(entry) = entry {
                             let filepath = entry.path();
                             if let Ok(metadata) = std::fs::symlink_metadata(&filepath) {
-                                let filename = if let Ok(fname) = filepath.strip_prefix(&cwd) {
-                                    fname
+                                let filename = if let Some(fname) = filepath.file_name() {
+                                    Path::new(fname)
                                 } else {
-                                    Path::new(&filepath)
+                                    Path::new(Path::new(&filepath).file_name().unwrap())
```

Comment:
To avoid the `unwrap`, you could do something similar to line 146, although I'll make another recommendation:

```
let fname = filepath
  .file_name()
  .or_else(|| Path::new(&filepath).file_name())
  .unwrap_or(filepath);

Path::new(fname)
```

Not 100% that'll work (not sure if the types line up), so you may have to tweak it.

Reply:
thanks for the tip